lib/core/service/firebase/firebase_service.dart

```diff
@@ -156,4 +163,38 @@ class FirebaseService {
     FirebaseMessaging.onMessageOpenedApp
         .listen(_firebaseMessagingBackgroundHandler);
   }
+
+  void addFcmToken() async  {
+    var tokenState = await getIt<AppOauth>().tokenState;
```

Comment:
we can use `tokenStateStream` to listen of tokenState changed and decide to addFcmToken or not

Reply:
Yep, sound good 👍